node/handleBlockRequests.go

```diff
@@ -39,23 +44,24 @@ func handleTransactionRequests(log zerolog.Logger, wg *sync.WaitGroup, net Netwo
 		case <-ticker.C:
 		}
 
-		requestMessages := getRequestMessages(requestsStream)
-		requestsQueue := newTransactionRequestsQueue(requestMessages)
+		requestMessages := getRequestMessages(peers, requestsStream)
+		requestsQueue := newBlockRequestsQueue(requestMessages)
```

Comment:
This is the part I can't follow. Why create a stream of requests that we can process and then go and turn it back into batches? In general, for Go, it can be said that:
1) Channels should mostly be used to avoid blocking when you have I/O. Using it to queue requests that can simply be processed immediately in local code (i.e. there is no network, disk, database I/O) should imo not make use of a channel in such a way.
2) When you do use channels, then you should just process things that come in as soon as possible, at the speed that works, rather than batching them again in this way.

Reply:
As mentioned above - I used the channel to model outgoing requests and every 100 ms or so, I am processing these. It doesn't necessary mean that call to net.Send has to block the processing of the next batch.

Reply:
The load balancing really shouldn't be done on each batch, because not every peer will process them at the same speed. There can be large differences due to network or hardware, so evenly distributing each batch, as discussed before, will create an uneven distribution and introduce unnecessary synchronization delays.